shared/settings/invites/container.js

```diff
@@ -4,52 +4,42 @@ import {connect} from 'react-redux'
 import Invites from './index'
 import {invitesReclaim, invitesRefresh, invitesSend, notificationsSave, notificationsToggle} from '../../actions/settings'
 
-import type {TypedState} from '../../constants/reducer'
 import type {Props} from './index'
+import type {TypedState} from '../../constants/reducer'
 
-type State = {
+export type RouteState = {
   email: string,
   message: string,
 }
 
-class InvitationsContainer extends Component<void, Props, State> {
-  state: State;
-
-  constructor (props: Props) {
-    super(props)
-    this.state = {
-      email: this.props.inviteEmail,
-      message: this.props.inviteMessage,
-    }
-  }
-
+class InvitationsContainer extends Component<void, Props, void> {
   componentWillMount () {
     this.props.onRefresh()
   }
 
-  onGenerateInvitation () {
-    const {email, message} = this.state
-    this.props.onGenerateInvitation(email, message)
-  }
-
   render () {
+    console.log(this.props)
     return <Invites
       {...this.props}
-      onChangeInviteEmail={email => this.setState({email})}
-      onChangeInviteMessage={message => this.setState({message})}
-      onGenerateInvitation={() => this.onGenerateInvitation()}
-      showMessageField={!!this.state.email}
+      showMessageField={!!this.props.inviteEmail}
     />
   }
 }
 
+// move to typedconnect
+// lenses that return where they came from in their composed stuff
```

Comment:
The mutterings of a madman... er... I guess these were notes I took when we were initially discussing lenses about a month ago. Whoops! Thanks for spotting them!